packages/babel-parser/test/fixtures/annex-b/enabled/assignmenttargettype/input.js

```diff
@@ -0,0 +1,12 @@
+function async() {}
+
+async() = 1;
+async() += 1;
+async()++;
+++async();
+for (async() in [1]) { }
+for (async() of [1]) {}
```

Comment:
Could you also add a test case for call expression nested within array pattern / object pattern? The `AssignmentTargetType` does not pass through the array / object pattern, so examples like `[f()] = []` should be rejected regardless of the annex B mode.